src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,21 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from '../../../shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | undefined;
+    @Output() buyClicked = new EventEmitter();
```

Comment:
Лучше стипизировать `EventEmitter` при помощи дженериков, чтобы было явно описано, что ожидать в `Output` событии
```suggestion
    @Output() buyClicked = new EventEmitter<string | undefined>();
```

В типизации `undefined` появляется из-за `?.` в 19 строке - `this.buyClicked.emit(this.product?._id); `

Reply:
спасибо, поправила!